### PR TITLE
Add npm workflow

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**
Add npm-publish workflow
Use diamondlightsource namespace
Remove empty test section from pypi action

### CODE REVIEW

File: .github/workflows/npm-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+      - run: npm ci
```
Comment (PeterC-DLS) [inline .github/workflows/npm-publish.yaml:18]: Should use pnpm instead

File: .github/workflows/npm-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
```
Comment (PeterC-DLS) [inline .github/workflows/npm-publish.yaml:7]: We want this to be triggered on a tag of a branch

File: client/example/package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@davidia/component": "workspace:^",
+    "@diamondlightsource/davidia": "workspace:^",
```
Comment (PeterC-DLS) [inline client/example/package.json:14]: There should be a corresponding fix in the example application to its imports

File: .github/workflows/npm-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  push:
+    tags:
+      - v**
```
Comment (PeterC-DLS) [inline .github/workflows/npm-publish.yaml:9]: A single asterisk will do. See [globs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

File: .github/workflows/npm-publish.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+
+jobs:
+  build:
```
Comment (PeterC-DLS) [inline .github/workflows/npm-publish.yaml:13]: Can be done with a single job

File: client/example/package.json (hunk 3 above)
Comment (PeterC-DLS) [reply client/example/package.json:14]: Right, it's in a separate module of a monorepo so should not use that relative path

File: .github/workflows/npm-publish.yaml (hunk 5 above)
Comment (VictoriaBeilsten-Edmands) [reply .github/workflows/npm-publish.yaml:13]: Get rid of `build` job and add `pnpm test` step to the publish job?

Comment (PeterC-DLS) [reply .github/workflows/npm-publish.yaml:13]: Yep!

File: .github/workflows/npm-publish.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    tags:
+      - v*
+
+
+jobs:
+  publish-npm:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (VictoriaBeilsten-Edmands) [reply .github/workflows/npm-publish.yaml]: Updated setup-node to v4 too.

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**